recipes/opencv/4.x/conanfile.py

```diff
@@ -1550,7 +1534,23 @@ def generate(self):
 
         if self.options.get_safe("with_wayland"):
             deps = PkgConfigDeps(self)
-            if not self._is_legacy_one_profile:
+            if self._is_legacy_one_profile:
+                # Manually generate pkgconfig file of wayland-protocols since
+                # PkgConfigDeps.build_context_activated can't work with legacy 1 profile
+                # We must use legacy conan v1 deps_cpp_info because self.dependencies doesn't
+                # contain build requirements when using 1 profile.
+                wp_prefix = self.deps_cpp_info["wayland-protocols"].rootpath
+                wp_version = self.deps_cpp_info["wayland-protocols"].version
```

Comment:
```suggestion
                wp_prefix = self.dependencies.build["wayland-protocols"].package_folder
                wp_version = self.dependencies.build["wayland-protocols"].ref.version
```

Dependencies.build is retro-compatible with Conan 1.x and will list wayland-protocol for sure.

Reply:
It's not retro compatible with conan v1 but not with 1 profile.

Reply:
So with 1 profile, `self.dependencies["wayland-protocols"]` is empty but not `self.dependencies.build["wayland-protocols"]`?

Reply:
Tool requires will be always available under `self.dependencies.build`, but not in `self.dependencies` directly. The number of profiles does not matter and the behavior is the same for Conan 1.x and 2.x

Reply:
ok, fixed by https://github.com/conan-io/conan-center-index/pull/21242/commits/9aad166fdedb76bd890662c5798d803b1a48703b